sklearn/metrics/tests/test_classification.py

```diff
@@ -1381,117 +1393,167 @@ def test_precision_recall_f1_no_labels_average_none():
     # |y_i| = [0, 0, 0]
     # |y_hat_i| = [0, 0, 0]
 
-    p, r, f, s = assert_warns(UndefinedMetricWarning,
-                              precision_recall_fscore_support,
-                              y_true, y_pred, average=None, beta=beta)
-    assert_array_almost_equal(p, [0, 0, 0], 2)
-    assert_array_almost_equal(r, [0, 0, 0], 2)
-    assert_array_almost_equal(f, [0, 0, 0], 2)
+    func = precision_recall_fscore_support
+    my_assert = (assert_warns if zero_division == "warn"
+                 else assert_no_warnings)
+    tmp = ([UndefinedMetricWarning, func] if zero_division == "warn"
+           else [func])
+    p, r, f, s = my_assert(*tmp, y_true, y_pred, average=None,
+                           beta=beta, zero_division=zero_division)
+    tmp = ([UndefinedMetricWarning, fbeta_score] if zero_division == "warn"
+           else [fbeta_score])
+    fbeta = my_assert(*tmp, y_true, y_pred, beta=beta,
+                      average=None, zero_division=zero_division)
+
+    zero_division = float(zero_division == 1)
```

Comment:
This is obfuscated. I'd rather a clear, separate test checking the behaviour of zero_divison, than a tiny, unexplicit piece in a larger test.

Reply:
simplified in the new version with two separated tests